DOCSP-13826 landing page (#17)
mongodb/docs-golang: source/index.txt

Introduction
------------

The official MongoDB Go Driver enables Go applications to connect to and
work with data in MongoDB. This page describes each section of the

> driver documentation. 

Reviewer: The "Learn" section is not a section of the docs, which might be confusing with this sentence here.

Author: Great catch! I changed it to a sub heading within the intro and changed the title to "Developer Hub" -- "Learn" itself was sort of misleading since there is no course on it

Learn
-----

Visit the Developer Hub for content, resources, community, and social
engagement for developers related to the Go driver:

> - To view articles on Go, see the `How To's and Articles page <https://www.mongodb.com/developer/learn/?content=Articles&text=Go>`__.

Reviewer: Question: why is this called How To's? I just see blog style posts and article, videos, podcasts at the bottom.

Author: It's the actual name of the tab when you hover over "Learn" at the top bar of the page

> - To view the forum on Go, see the `forums page <https://www.mongodb.com/community/forums/tag/golang>`__.

Reviewer: ```suggestion
- To view the Go forum, see the `forums page <https://www.mongodb.com/community/forums/tag/golang>`__.
```

Quick Start
-----------

To get up and running quickly, take a look at the :doc:`Quick Start
</quick-start>` to learn how to establish a connection to MongoDB Atlas
and begin to work with data.

Usage Examples
--------------

The :doc:`Usage Examples </usage-examples>` section provides
runnable code snippets and explanations for common methods. We recommend
this section for users that are new to the MongoDB Go Driver.

Fundamentals
------------

> Whether you are new to MongoDB or you just want to brush up on the core

Reviewer: ```suggestion
Whether you are new to MongoDB or want to brush up on core
```

concepts, you can check out the **Fundamentals** section which covers
the following material in the Go driver:

- :doc:`Connect to MongoDB </fundamentals/connection>`
- :doc:`Context Usage in MongoDB </fundamentals/context>`
- :doc:`Authenticate with MongoDB </fundamentals/auth>`

.. - :doc:`Specify an API Version </fundamentals/versioned-api>`
.. - :doc:`Use the Driver's Data Formats </fundamentals/data-formats>`
.. - :doc:`Perform CRUD Operations </fundamentals/crud>`
.. - :doc:`Perform Aggregations </fundamentals/aggregation>`
.. - :doc:`Construct Indexes </fundamentals/indexes>`
.. - :doc:`Specify Collations </fundamentals/collations>`
.. - :doc:`Record Events in the Driver </fundamentals/logging>`
.. - :doc:`Use Driver Events in your Code </fundamentals/monitoring>`
.. - :doc:`Store and Retrieve Files in MongoDB </fundamentals/gridfs>`

API
---

See the `API Documentation
<https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo>`__ if you are
looking for technical information about classes, methods, and

> configuration objects within the MongoDB Go Driver.

Reviewer: I think Go calls them configuration structs, but that could be too jargon-y for the landing page. is there another way to describe this? Types and functions?

Author: Agreed! Types and functions sounds the most accurate.

.. FAQ
.. ---

.. The :doc:`Frequently Asked Questions (FAQ) </faq>` section provides answers
.. to commonly asked questions about the MongoDB Go Driver.

Issues & Help
-------------

The :doc:`Issues & Help </issues-and-help>` section explains how to report
bugs, contribute to the driver, and find additional resources for asking
questions and receiving help.

Compatibility
-------------

> The :doc:`Compatibility </compatibility>` sections compatibility charts
> show the recommended Go driver version for each MongoDB server version. 

Reviewer: ```suggestion
The :doc:`Compatibility </compatibility>` section contains compatibility charts
that show the recommended Go Driver version for each MongoDB Server version. 
```

Introduction
------------

The official MongoDB Go Driver enables Go applications to connect to and

> work with data in MongoDB. On this page, you can find descriptions of

Reviewer 2: Suggestion: simplify the last sentence:

`This page describes each section of the driver documentation.`

each section of the driver documentation.

Learn
-----

> Visit the Developer Hub for content, resources, community, and social

Reviewer 2: Nit: Since the links below are part of the Developer Hub (as opposed to separate resources), I would use a bulleted list.

```
Visit the Developer Hub for content, resources, community, and social engagement for developers related to the Go driver:

- To view articles on Go, see the `How To's and Articles page
<https://www.mongodb.com/developer/learn/?content=Articles&text=Go>`_.
- To view the forum on Go, see the `forums page
<https://www.mongodb.com/community/forums/tag/golang>`_.
```

engagement for developers related to the Go driver.

To view articles on Go, see the `How To's and Articles page
<https://www.mongodb.com/developer/learn/?content=Articles&text=Go>`_.

To view the forum on Go, see the `forums page
<https://www.mongodb.com/community/forums/tag/golang>`_.

Quick Start
-----------

To get up and running quickly, take a look at the :doc:`Quick Start

> </quick-start>` to learn how to establish a connection to MongoDB Atlas

Reviewer 2: Nit: I would be more specific than establishing a connection and would break this into 2 sentences:

```
To get up and running quickly, take a look at the :doc:`Quick Start
</quick-start>`, which shows you how to: 

- Create an application,
- Use the Go driver to connect to MongoDB Atlas, and
- Begin to work with data.
```

and begin to work with data.

Usage Examples
--------------

The :doc:`Usage Examples </usage-examples>` section provides
runnable code snippets and explanations for common methods. We recommend
this section for users that are new to the MongoDB Go Driver.

Fundamentals
------------

Whether you are new to MongoDB or you just want to brush up on the core

> concepts, you can check out the **Fundamentals** section which contains

Reviewer 2: Nit: simplify language a little:

```
you can check out the **Fundamentals** section, which reviews:

- :doc:`Connecting to MongoDB with the Go driver </fundamentals/connection>`
- :doc:`Context Usage in MongoDB </fundamentals/context>`
- :doc:`Authentication with MongoDB </fundamentals/auth>`
```

.. - :doc:`Use the Driver's Data Formats </fundamentals/data-formats>`
.. - :doc:`Perform CRUD Operations </fundamentals/crud>`
.. - :doc:`Perform Aggregations </fundamentals/aggregation>`
.. - :doc:`Construct Indexes </fundamentals/indexes>`
.. - :doc:`Specify Collations </fundamentals/collations>`
.. - :doc:`Record Events in the Driver </fundamentals/logging>`
.. - :doc:`Use Driver Events in your Code </fundamentals/monitoring>`
.. - :doc:`Store and Retrieve Files in MongoDB </fundamentals/gridfs>`

API
---

See the `API Documentation
<https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo>`__ if you are
looking for technical information about classes, methods, and
configuration objects within the MongoDB Go Driver.

.. FAQ
.. ---

.. The :doc:`Frequently Asked Questions (FAQ) </faq>` section provides answers
.. to commonly asked questions about the MongoDB Go Driver.

Issues & Help
-------------

The :doc:`Issues & Help </issues-and-help>` section explains how to report
bugs, contribute to the driver, and find additional resources for asking
questions and receiving help.

Compatibility
-------------

The :doc:`Compatibility </compatibility>` section shows compatibility charts

> with Go and MongoDB server versions.

Reviewer 2: Wording is a little awkward, I would change to:

`The :doc:`Compatibility </compatibility>` section's compatibility charts show the recommended Go driver version for each MongoDB server version.`
